### ISSUE TITLE
How to see better performance in knowledge distillation from Llama3.2-3B to 1B?

### ISSUE THREAD

**Issue by asmitamohanty** [pytorch/torchtune#2922]
I am new to torchtune & wanted to distill from Llama 3.2-3B (teacher) to Llama 3.2-1B (student). I was following this tutorial - https://docs.pytorch.org/torchtune/0.3/tutorials/llama_kd_tutorial.html . 

I know that finetuning teacher will help in improving performance, but just by considering the baselines, atleast I should see the trend where the new distilled student model (KD output) performs better than the baseline student model? 

I am seeing the distilled model performing the worst. I am sort of unable to replicate the results from the tutorial though they're using a different kd config which is not present in the latest torchtune git version. 

<details><summary>Baseline Teacher</summary>
<p>

|    Tasks     |Version|Filter|n-shot| Metric |   |Value |   |Stderr|
|--------------|-------|------|------|--------|---|-----:|---|-----:|
|arc_challenge |      1|none  |None  |acc     |↑  |0.4352|±  |0.0145|
|              |       |none  |None  |acc_norm|↑  |0.4608|±  |0.0146|
|arc_easy      |      1|none  |None  |acc     |↑  |0.7445|±  |0.0089|
|              |       |none  |None  |acc_norm|↑  |0.6785|±  |0.0096|
|commonsense_qa|Yaml   |none  |None  |acc     |↑  |0.6749|±  |0.0134|
|hellaswag     |      1|none  |None  |acc     |↑  |0.5213|±  |0.0050|
|              |       |none  |None  |acc_norm|↑  |0.7046|±  |0.0046|
|truthfulqa_mc2|      2|none  |0     |acc     |↑  |0.4972|±  |0.0148|

</p>
</details> 

<details><summary>Baseline Student</summary>
<p>

|    Tasks     |Version|Filter|n-shot| Metric |   |Value |   |Stderr|
|--------------|-------|------|------|--------|---|-----:|---|-----:|
|arc_challenge |      1|none  |None  |acc     |↑  |0.3584|±  |0.0140|
|              |       |none  |None  |acc_norm|↑  |0.3814|±  |0.0142|
|arc_easy      |      1|none  |None  |acc     |↑  |0.6852|±  |0.0095|
|              |       |none  |None  |acc_norm|↑  |0.6313|±  |0.0099|
|commonsense_qa|Yaml   |none  |None  |acc     |↑  |0.5545|±  |0.0142|
|hellaswag     |      1|none  |None  |acc     |↑  |0.4516|±  |0.0050|
|              |       |none  |None  |acc_norm|↑  |0.6074|±  |0.0049|
|truthfulqa_mc2|      2|none  |0     |acc     |↑  |0.4385|±  |0.0144|

</p>
</details> 

<details><summary>KD model</summary>
<p>

|    Tasks     |Version|Filter|n-shot| Metric |   |Value |   |Stderr|
|--------------|-------|------|------|--------|---|-----:|---|-----:|
|arc_challenge |      1|none  |None  |acc     |↑  |0.2244|±  |0.0122|
|              |       |none  |None  |acc_norm|↑  |0.2483|±  |0.0126|
|arc_easy      |      1|none  |None  |acc     |↑  |0.2652|±  |0.0091|
|              |       |none  |None  |acc_norm|↑  |0.2837|±  |0.0092|
|commonsense_qa|Yaml   |none  |None  |acc     |↑  |0.2072|±  |0.0116|
|hellaswag     |      1|none  |None  |acc     |↑  |0.2795|±  |0.0045|
|              |       |none  |None  |acc_norm|↑  |0.3030|±  |0.0046|
|truthfulqa_mc2|      2|none  |0     |acc     |↑  |0.4722|±  |0.0164|

</p>
</details> 

My config file is this:
```swift
output_dir: /tmp/torchtune/llama3_2_3B_to_1B/KD_lora_single_device # /tmp may be deleted by your system. Change it to your preference.

# Model Arguments
model:
  _component_: torchtune.models.llama3_2.llama3_2_1b

teacher_model:
  _component_: torchtune.models.llama3_2.llama3_2_3b

# Tokenizer
tokenizer:
  _component_: torchtune.models.llama3.llama3_tokenizer
  path: /tmp/Llama-3.2-1B-Instruct/original/tokenizer.model
  max_seq_len: null

checkpointer:
  _component_: torchtune.training.FullModelHFCheckpointer
  checkpoint_dir: /tmp/Llama-3.2-1B-Instruct/
  checkpoint_files: [
    model.safetensors
  ]
  recipe_checkpoint: null
  output_dir: ${output_dir}
  model_type: LLAMA3
resume_from_checkpoint: False
save_adapter_weights_only: False

# Teacher checkpoint
teacher_checkpointer:
  _component_: torchtune.training.FullModelHFCheckpointer
  checkpoint_dir: /tmp/Meta-Llama-3.2-3B-Instruct/
  checkpoint_files: [
    model-00001-of-00002.safetensors,
    model-00002-of-00002.safetensors
  ]
  recipe_checkpoint: null
  output_dir: ${output_dir}
  model_type: LLAMA3

# Dataset and Sampler
dataset:
  _component_: torchtune.datasets.alpaca_cleaned_dataset
  packed: False  # True increases speed
seed: null
shuffle: True
batch_size: 4

# Optimizer and Scheduler
optimizer:
  _component_: torch.optim.AdamW
  fused: True
  weight_decay: 0.01
  lr: 3e-4
lr_scheduler:
  _component_: torchtune.training.lr_schedulers.get_cosine_schedule_with_warmup
  num_warmup_steps: 100

loss:
  _component_: torchtune.modules.loss.CEWithChunkedOutputLoss

kd_loss:
  _component_: torchtune.modules.loss.ForwardKLWithChunkedOutputLoss
kd_ratio: 0.5

# Training
epochs: 1
max_steps_per_epoch: null
gradient_accumulation_steps: 8  # Use to increase effective batch size
clip_grad_norm: null
compile: False  # torch.compile the model + loss, True increases speed + decreases memory

# Logging
metric_logger:
  _component_: torchtune.training.metric_logging.DiskLogger
  log_dir: ${output_dir}/logs
log_every_n_steps: 1
log_peak_memory_stats: True
log_level: INFO  # DEBUG, WARN, etc.

# Environment
device: cuda
dtype: bf16

# Activations Memory
enable_activation_checkpointing: False  # True reduces memory
enable_activation_offloading: False  # True reduces memory



# Profiler (disabled)
profiler:
  _component_: torchtune.training.setup_torch_profiler
  enabled: False

  #Output directory of trace artifacts
  output_dir: ${output_dir}/profiling_outputs

  #`torch.profiler.ProfilerActivity` types to trace
  cpu: True
  cuda: True

  #trace options passed to `torch.profiler.profile`
  profile_memory: False
  with_stack: False
  record_shapes: True
  with_flops: False

  # `torch.profiler.schedule` options:
  # wait_steps -> wait, warmup_steps -> warmup, active_steps -> active, num_cycles -> repeat
  wait_steps: 5
  warmup_steps: 3
  active_steps: 2
  num_cycles: 1
```
